Enhance mikrotik system with error return and graphs

Sephtex opened the pull request:
This PR adds an error catching function, that will return the error of SSH in case rc != 0.

In addition:
* I forced the numbers of disk/memory to integers since the graphs where just returning -nan.
* Changed range of bad blocks to max 5% as the mikrotik gets broken on reboot when going over 5% and it needs to be reinstalled to get it back to 0

I'm currently testing, over a few hours I should see a better result

Sephtex commented:
mmm by using integers in the _memory and _cpu graphs, the graphs are still displaying -nan :thinking:

quotengrote commented:
> mmm by using integers in the _memory and _cpu graphs, the graphs are still displaying -nan 🤔

Interesting, here it works:
![image](https://user-images.githubusercontent.com/38253905/143686374-41c987c0-c7f6-4eaa-a914-1d1ab5ef9878.png)

Sephtex commented:
It's strange, I tried to adapt what I felt might be a possible problem. Yet non seemed to help, the graphs are still empty:
![image](https://user-images.githubusercontent.com/25793199/143754779-64981473-b2ad-48a2-8a50-d509091ae8d7.png)
![image](https://user-images.githubusercontent.com/25793199/143755011-f6846882-e86a-4fe6-8fbd-f2a3d09839bc.png)

Data returned, where I don't see anything wrong either:
```
$ munin-run mikrotik_system config
multigraph voltage_graph_Sephtex
graph_title voltage Sephtex
graph_vlabel volt
graph_category mikrotik
graph_args -l 0
voltage.label voltage
graph_info Input Voltage.
multigraph bad_blocks_graph_Sephtex
graph_title bad blocks Sephtex
graph_vlabel %
graph_category mikrotik
graph_args -l 0 --upper-limit 5
bad_blocks.label bad_blocks
bad_blocks.warning 3
bad_blocks.critical 4
graph_info Percentage of Bad Blocks.
multigraph write_sect_total_graph_Sephtex
graph_title Total sector writes Sephtex
graph_vlabel count
graph_category mikrotik
graph_args -l 0
write_sect_total.label write_sect_total
graph_info Total sector writes.
multigraph write_sect_since_reboot_graph_Sephtex
graph_title Sector writes since reboot Sephtex
graph_vlabel count
graph_category mikrotik
graph_args -l 0
write_sect_since_reboot.label write_sect_since_reboot
graph_info Total Sector writes since last reboot.
multigraph temperature_graph_Sephtex
graph_title temperature Sephtex
graph_vlabel °C
graph_category mikrotik
graph_args -l 0
temperature.label cpu temperature
temperature.warning 75
temperature.critical 90
multigraph cpu_load_graph_Sephtex
graph_title cpu load Sephtex
graph_vlabel %
graph_category mikrotik
graph_args -l 0 --upper-limit 100
cpu_total.label total load
cpu_total.warning 75
cpu_total.critical 90
graph_info Total CPU Load and Load, IRQ and Disk per Core.
cpu0_load.label cpu0 load
cpu0_irq.label cpu0 irq
cpu0_disk.label cpu0 disk
multigraph memory_graph_Sephtex
graph_title memory Sephtex
graph_vlabel MiB
graph_category mikrotik
graph_args -l 0
total_memory.label total memory
used_memory.label used memory
free_memory.label free memory
graph_info Total, Used & free RAM.
used_memory.critical 200
used_memory.warning 155
multigraph disk_graph_Sephtex
graph_title disk Sephtex
graph_vlabel Bytes
graph_category mikrotik
graph_args -l 0
total_disk.label total disk space
free_disk.label free disk space
graph_info Total & free disk space.

$ munin-run mikrotik_system
multigraph voltage_graph_Sephtex
voltage.value 23.2
multigraph bad_blocks_graph_Sephtex
bad_blocks.value 0.8
multigraph write_sect_total_graph_Sephtex
write_sect_total.value 36947339
multigraph write_sect_since_reboot_graph_Sephtex
write_sect_since_reboot.value 4898267
multigraph temperature_graph_Sephtex
temperature.value 50
multigraph cpu_load_graph_Sephtex
cpu_total.value 34
cpu0_load.value 34
cpu0_irq.value 33
cpu0_disk.value 0
multigraph memory_Sephtex
total_memory.value 256
free_memory.value 221
used_memory.value 34
multigraph disk_Sephtex
free_disk.value 103075020
total_disk.value 134217728
```

quotengrote commented:
Very confusing, the data returned is an int as expected by munin.

Could you try to delete the rrd-files and let munin create them new?

Sephtex commented:
> Very confusing, the data returned is an int as expected by munin.
> 
> Could you try to delete the rrd-files and let munin create them new?

Just done it, including the munin async files for the module. Lets see if it'll generate the graphs!

Sephtex commented:
I'm clueless, I updated even from alpine-3.13 to 3.14 on the master node to have 2.0.67-r0 with the hopes it would change something, but it didn't. What I noticed on the files where:
```
bash-5.1# cd /var/lib/munin
bash-5.1# find . -name '*Sephtex*' | xargs ls -la
-rw-r--r-- 1 munin munin 50824 Nov 28 18:55 ./sephtex-host/kvm-dev.pr2-nuc.sephtex.net-bad_blocks_graph_Sephtex-bad_blocks-g.rrd
-rw-r--r-- 1 munin munin 50824 Nov 28 18:55 ./sephtex-host/kvm-dev.pr2-nuc.sephtex.net-cpu_load_graph_Sephtex-cpu0_disk-g.rrd
-rw-r--r-- 1 munin munin 50824 Nov 28 18:55 ./sephtex-host/kvm-dev.pr2-nuc.sephtex.net-cpu_load_graph_Sephtex-cpu0_irq-g.rrd
-rw-r--r-- 1 munin munin 50824 Nov 28 18:55 ./sephtex-host/kvm-dev.pr2-nuc.sephtex.net-cpu_load_graph_Sephtex-cpu0_load-g.rrd
-rw-r--r-- 1 munin munin 50824 Nov 28 18:55 ./sephtex-host/kvm-dev.pr2-nuc.sephtex.net-cpu_load_graph_Sephtex-cpu_total-g.rrd
-rw-r--r-- 1 munin munin 50824 Nov 28 18:55 ./sephtex-host/kvm-dev.pr2-nuc.sephtex.net-disk_Sephtex-free_disk-g.rrd
-rw-r--r-- 1 munin munin 50824 Nov 28 18:55 ./sephtex-host/kvm-dev.pr2-nuc.sephtex.net-disk_Sephtex-total_disk-g.rrd
-rw-r--r-- 1 munin munin 50824 Nov 28 13:00 ./sephtex-host/kvm-dev.pr2-nuc.sephtex.net-disk_graph_Sephtex-free_disk-g.rrd
-rw-r--r-- 1 munin munin 50824 Nov 28 13:00 ./sephtex-host/kvm-dev.pr2-nuc.sephtex.net-disk_graph_Sephtex-total_disk-g.rrd
-rw-r--r-- 1 munin munin 50824 Nov 28 18:55 ./sephtex-host/kvm-dev.pr2-nuc.sephtex.net-memory_Sephtex-free_memory-g.rrd
-rw-r--r-- 1 munin munin 50824 Nov 28 18:55 ./sephtex-host/kvm-dev.pr2-nuc.sephtex.net-memory_Sephtex-total_memory-g.rrd
-rw-r--r-- 1 munin munin 50824 Nov 28 18:55 ./sephtex-host/kvm-dev.pr2-nuc.sephtex.net-memory_Sephtex-used_memory-g.rrd
-rw-r--r-- 1 munin munin 50824 Nov 28 13:00 ./sephtex-host/kvm-dev.pr2-nuc.sephtex.net-memory_graph_Sephtex-free_memory-g.rrd
-rw-r--r-- 1 munin munin 50824 Nov 28 13:00 ./sephtex-host/kvm-dev.pr2-nuc.sephtex.net-memory_graph_Sephtex-total_memory-g.rrd
-rw-r--r-- 1 munin munin 50824 Nov 28 13:00 ./sephtex-host/kvm-dev.pr2-nuc.sephtex.net-memory_graph_Sephtex-used_memory-g.rrd
-rw-r--r-- 1 munin munin 50824 Nov 28 18:55 ./sephtex-host/kvm-dev.pr2-nuc.sephtex.net-temperature_graph_Sephtex-temperature-g.rrd
-rw-r--r-- 1 munin munin 50824 Nov 28 18:55 ./sephtex-host/kvm-dev.pr2-nuc.sephtex.net-voltage_graph_Sephtex-voltage-g.rrd
-rw-r--r-- 1 munin munin 50824 Nov 28 18:55 ./sephtex-host/kvm-dev.pr2-nuc.sephtex.net-write_sect_since_reboot_graph_Sephtex-write_sect_since_reboot-g.rrd
-rw-r--r-- 1 munin munin 50824 Nov 28 18:55 ./sephtex-host/kvm-dev.pr2-nuc.sephtex.net-write_sect_total_graph_Sephtex-write_sect_total-g.rrd
```
```
# ls -al *mikrotik_system*
-rw-r--r-- 1 munin-async munin  2567 Nov 28 12:59 munin-daemon.mikrotik_system.1638097200.3600
-rw-r--r-- 1 munin-async munin 30804 Nov 28 13:59 munin-daemon.mikrotik_system.1638100800.3600
-rw-r--r-- 1 munin-async munin 30804 Nov 28 14:59 munin-daemon.mikrotik_system.1638104400.3600
-rw-r--r-- 1 munin-async munin 30804 Nov 28 15:59 munin-daemon.mikrotik_system.1638108000.3600
-rw-r--r-- 1 munin-async munin 30804 Nov 28 16:59 munin-daemon.mikrotik_system.1638111600.3600
-rw-r--r-- 1 munin-async munin 30804 Nov 28 17:59 munin-daemon.mikrotik_system.1638115200.3600
-rw-r--r-- 1 munin-async munin 30804 Nov 28 18:59 munin-daemon.mikrotik_system.1638118800.3600
```
What I notice is that the rrd files never seem to be updated since the initial time for the 2 multigraphs.
Although I notice the following in munin-run output:
```
multigraph memory_graph_Sephtex
graph_title memory Sephtex
graph_vlabel MiB
graph_category mikrotik
graph_args -l 0
total_memory.label total memory
used_memory.label used memory
free_memory.label free memory
graph_info Total, Used & free RAM.
used_memory.critical 202
used_memory.warning 157
multigraph disk_graph_Sephtex
graph_title disk Sephtex
graph_vlabel Bytes
graph_category mikrotik
graph_args -l 0
total_disk.label total disk space
free_disk.label free disk space
graph_info Total & free disk space.
```
```
multigraph memory_Sephtex <- _graph seems to be missing here
total_memory.value 256
free_memory.value 224
used_memory.value 32
multigraph disk_Sephtex <- _graph seems to be missing here
free_disk.value 103075020
total_disk.value 134217728
```
I updated it in the script, lets see later!

quotengrote commented:
That could be the error, but why is it working here?

Values:
```
mg@docker2 ~/docker/docker-munin-master (master)                                                                                                                            > sudo munin-run mt_system_rb5009                                                                                                                                           multigraph bad_blocks_graph_rb5009                                                                                                                                          bad_blocks.value 0                                                                                                                                                          multigraph write_sect_total_graph_rb5009                                                                                                                                    write_sect_total.value 11032                                                                                                                                                multigraph write_sect_since_reboot_graph_rb5009                                                                                                                             write_sect_since_reboot.value 11032                                                                                                                                         multigraph temperature_graph_rb5009                                                                                                                                         temperature.value 46                                                                                                                                                        multigraph cpu_load_graph_rb5009                                                                                                                                            cpu_total.value 0                                                                                                                                                           cpu0_load.value 0                                                                                                                                                           cpu0_irq.value 0                                                                                                                                                            cpu0_disk.value 0                                                                                                                                                           cpu1_load.value 0                                                                                                                                                           cpu1_irq.value 0                                                                                                                                                            cpu1_disk.value 0                                                                                                                                                           cpu2_load.value 2                                                                                                                                                           cpu2_irq.value 0                                                                                                                                                            cpu2_disk.value 0                                                                                                                                                           cpu3_load.value 0
cpu3_irq.value 0
cpu3_disk.value 0
multigraph memory_rb5009
total_memory.value 1024.0
free_memory.value 823.9
used_memory.value 200.1
multigraph disk_rb5009
free_disk.value 1.04134e+09
total_disk.value 1074790400
```

Config:
```
mg@docker2 ~/docker/docker-munin-master (master)
> sudo munin-run mt_system_rb5009 config
multigraph bad_blocks_graph_rb5009
graph_title bad blocks rb5009
graph_vlabel %
graph_category mikrotik
graph_args -l 0 --upper-limit 100
bad_blocks.label bad_blocks
bad_blocks.warning 50
bad_blocks.critical 90
graph_info Percentage of Bad Blocks.
multigraph write_sect_total_graph_rb5009
graph_title Total sector writes rb5009
graph_vlabel count
graph_category mikrotik
graph_args -l 0
write_sect_total.label write_sect_total
graph_info Total sector writes.
multigraph write_sect_since_reboot_graph_rb5009
graph_title Sector writes since reboot rb5009
graph_vlabel count
graph_category mikrotik
graph_args -l 0
write_sect_since_reboot.label write_sect_since_reboot
graph_info Total Sector writes since last reboot.
multigraph temperature_graph_rb5009
graph_title temperature rb5009
graph_vlabel °C
graph_category mikrotik
graph_args -l 0
temperature.label cpu temperature
temperature.warning 75
temperature.critical 90
multigraph cpu_load_graph_rb5009
graph_title cpu load rb5009
graph_vlabel %
graph_category mikrotik
graph_args -l 0 --upper-limit 100
cpu_total.label total load
cpu_total.warning 75
cpu_total.critical 90
graph_info Total CPU Load and Load, IRQ and Disk per Core.
cpu0_load.label cpu0 load
cpu0_irq.label cpu0 irq
cpu0_disk.label cpu0 disk
cpu1_load.label cpu1 load
cpu1_irq.label cpu1 irq
cpu1_disk.label cpu1 disk
cpu2_load.label cpu2 load
cpu2_irq.label cpu2 irq
cpu2_disk.label cpu2 disk
cpu3_load.label cpu3 load
cpu3_irq.label cpu3 irq
cpu3_disk.label cpu3 disk
multigraph memory_graph_rb5009
graph_title memory rb5009
graph_vlabel MiB
graph_category mikrotik
graph_args -l 0
total_memory.label total memory
used_memory.label used memory
free_memory.label free memory
graph_info Total, Used & free RAM.
used_memory.critical 741.51
used_memory.warning 576.73
multigraph disk_graph_rb5009
graph_title disk rb5009
graph_vlabel Bytes
graph_category mikrotik
graph_args -l 0
total_disk.label total disk space
free_disk.label free disk space
graph_info Total & free disk space.
```

Sephtex commented:
I have no clue @quotengrote , but with my last commit, it seems to have fixed it:
![image](https://user-images.githubusercontent.com/25793199/143777715-b3eeaceb-0a2a-402b-a564-58c1738d12fe.png)
So for me the changes are ok, unless you want me to revert forcing to integers?

Maybe @sumpfralle knows why it is working for you? Maybe it's a symptom and a bug in munin-async (since I'm using async)?

sumpfralle commented:
From my point of view, I would merge it now.

I did not follow your exchange regarding data type problems in detail. If you think, there is an issue, then please open a separate issue.

If anyone feels like: adding a bit more documentation: the nested implementation of the `catch` function is quite nifty - I think, it could benefit a lot from a bit of documentation/comments :)

Anyway: just tell me, when I should merge it.

Sephtex commented:
@sumpfralle as a summary: I have no clue why it is working for @quotengrote and not for me with the mismatched multigraph name. Only guess I could have is that I'm using munin-async where maybe the issue can be detected.

Anyway, the `catch` function is the best solution I found on stackoverflow and just copy-pasta it :-). I tried a few other ways how to be able to get the stdout/stderr into a separate variable, but this was the best I could find after some testing.

On the last commit, I added the source of the catch function where it is explained in very detail, hope it will be enough!

For me it would be great to have it merged by the time I'll roll it out on all my other systems, unless someone else has a remark?

Sephtex commented:
btw @quotengrote , I noticed that the name you had for your routerboard is rb5009, is that the name that you gave to the RouterOS? or is it fetched from board-name?

When I execute the function to get the name I get the following which is then just catch the last line with the awk and returning of the subshell:
```
sshpass -p  "[REDACTED]" ssh -o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no "munin-dev"@"192.168.2.1" ':delay 6s; /system health print; /system resource print; /system resource cpu print; /system identity print' | grep name:
Warning: Permanently added '192.168.2.1' (RSA) to the list of known hosts.
        architecture-name: mipsbe
               board-name: RB493G
  name: Sephtex router
```
And for me it resulted in Sephtex, but I'll make a commit to get 'Sephtex_router', unless you did want to fetch the board-name which I don't think that would be the case?

This can be a breaking change for the users using the module and replacing with this updated version. Is that ok?

Sephtex commented:
made the changes in 4e489497, hope you agree with that @quotengrote , else let me know or I'll revert!

quotengrote commented:
I would like to try i tomorrow before it gets merged if that is possible.

> Am 28.11.2021 um 19:50 schrieb Younes Ichiche ***@***.***>:
> 
> ﻿
> @sumpfralle as a summary: I have no clue why it is working for @quotengrote and not for me with the mismatched multigraph name. Only guess I could have is that I'm using munin-async where maybe the issue can be detected.
> 
> Anyway, the catch function is the best solution I found on stackoverflow and just copy-pasta it :-). I tried a few other ways how to be able to get the stdout/stderr into a separate variable, but this was the best I could find after some testing.
> 
> On the last commit, I added the source of the catch function where it is explained in very detail, hope it will be enough!
> 
> For me it would be great to have it merged by the time I'll roll it out on all my other systems, unless someone else has a remark?
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub, or unsubscribe.

quotengrote commented:
tjx, will take a look tomorrow

> Am 28.11.2021 um 20:13 schrieb Younes Ichiche ***@***.***>:
> 
> ﻿
> made the changes in 4e48949, hope you agree with that @quotengrote , else let me know or I'll revert!
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub, or unsubscribe.

quotengrote commented:
> btw @quotengrote , I noticed that the name you had for your routerboard is rb5009, is that the name that you gave to the RouterOS? or is it fetched from board-name?

It is the name set in ROS.

> 
> When I execute the function to get the name I get the following which is then just catch the last line with the awk and returning of the subshell:
> 
> ```
> sshpass -p  "[REDACTED]" ssh -o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no "munin-dev"@"192.168.2.1" ':delay 6s; /system health print; /system resource print; /system resource cpu print; /system identity print' | grep name:
> Warning: Permanently added '192.168.2.1' (RSA) to the list of known hosts.
>         architecture-name: mipsbe
>                board-name: RB493G
>   name: Sephtex router
> ```
You are right, didnt see the  `architecture-name` and `board-name`-value. 
But that could be fixed with:
Line: 
`name=$(echo "$line" | grep name: | awk '{ print $2 }')` --> `name=$(echo "$line" | awk '/\sname:/{ print $2}')`


> This can be a breaking change for the users using the module and replacing with this updated version. Is that ok?

Maybe we could check if the name is set manually and than use it, and if it is not set use the name supplied by `/system identity print`?

sumpfralle commented:
The latest comment sounds like I should not merge it, yet?

quotengrote commented:
That's right, i would try to fix that. But probably but not before the weekend.

> Am 30.11.2021 um 20:21 schrieb Lars Kruse ***@***.***>:
> 
> ﻿
> The latest comment sounds like I should not merge it, yet?
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub, or unsubscribe.

Sephtex commented:
I tried to adapt the get_name function but for some reason the result is always empty when I test it:
```
function get_name {
  local fname
  while read -r line; do
    if echo "$line" | grep -qE '(board-)?name:'; then
      fname="$(echo "$line" | awk '/\s+(board-)?name:/{ print $2 }')"
      if [ -n "$fname" ]; then
        name="$fname"
      fi
    fi
  done <<< "$data"
}
```
I'll check later when I have some moment unless you might know why.

In addition, I don't know why you have the awk error since that doesn't pop up for me.
Although I'm thinking about character encoding, as I see some German signs not displaying correctly, could it be it messes with the script?
![image](https://user-images.githubusercontent.com/25793199/144589695-47f24f01-8f08-40ab-bbe7-fb4d2c056699.png)
Maybe you could help in translating them to English? :)

quotengrote commented:
Maybe you could help in translating them to English? :)

Can do, should i push it to the same branch?

Regarding the name-Function, i will take a look hopefully that weekend.

Sephtex commented:
I'm not sure what the best practices are on github for this kind of situation? I would think that a PR would be created towards my branch (starting from my branch and make the additional changes)?

⁣Get BlueMail for Android ​

On 3 Dec 2021, 12:15, at 12:15, Michael Grote ***@***.***> wrote:
>> Maybe you could help in translating them to English? :)
>
>Can do, should i push it to the same branch?
>
>Regarding the name-Function, i will take a look hopefully that weekend.
>
>-- 
>You are receiving this because you authored the thread.
>Reply to this email directly or view it on GitHub:
>https://github.com/munin-monitoring/contrib/pull/1262#issuecomment-985435138

quotengrote commented:
Ok, i will make a PR towards your Repo

quotengrote commented:
PR is created: https://github.com/Sephtex/contrib/pull/2

Sephtex commented:
Hi @quotengrote ,
Can you check now with the changed get_name function? Unfortunately the awk solution you gave didn't worked as the name was then prepended and appended with a `_` (or lets say a space). I also needed to use `for` instead of a `while`-loop as the `while` was executed in a subshell casing my variable to be empty afterwards, which was the reason for it to fail earlier.

If you have a way how to get the trimming and replacing of any other special character to `_`, feel free to propose! But note that the mikrotik returns some strange end line character that gave some strange result. (like only having an \r)

quotengrote commented on plugins/router/mikrotik_system:
```diff
@@ -133,10 +162,10 @@ function get_bad_blocks_label {
       print "bad_blocks.label bad_blocks";
-      print "bad_blocks.warning 50";
-      print "bad_blocks.critical 90";
+      print "bad_blocks.warning 3;
```
The closing " is missing.

Sephtex replied:
oops, fixed :)

quotengrote commented:
But other than that it works flawlessly til now.

quotengrote commented:
Hmm, the temperature does not get reported anymore.
![image](https://user-images.githubusercontent.com/38253905/144743243-49b5d44e-9998-4f14-9a73-1a6dbcb103be.png)

with munin-run mt_system_rb5009 the config values for temperatures get reported but not the values. But that part wasnt touched...

quotengrote commented:
The problems seems to be in the `get_ros-version` Function, it reports Version 6, if i set the variable directly it works.

Sephtex commented:
Oh, for me it is still working as expected:
![image](https://user-images.githubusercontent.com/25793199/144743643-3abaf440-80e4-477f-b157-27052853cb18.png)
Although, for me it's the memory that doesn't seem to report anything now:
![image](https://user-images.githubusercontent.com/25793199/144743684-37e076c9-e1a8-4449-b288-61eba05178b1.png)

quotengrote commented:
Thats interesting, memory works for me...

quotengrote commented:
Could you try it the "new" get_ros_version works for you?

```
function get_ros_version {
  while read -r line; do
    if echo "$line" | grep -q 'version: '; then
      if echo "$line" | awk '/version:/{print $2}' | grep -q  "^7."; then
        ros_version="7"
      else
        ros_version="6"
      fi
    fi
  done <<< "$data"
}
```
For me it works again.

sumpfralle commented:
> `> if echo "$line" | awk '/version:/{print $2}' | grep -q  "^7."; then`

I would recommend to use something like the following instead:
```shell
if echo "$line" | awk '/version:/{print $2}' | grep -q  "^[0-6]\."; then
```

Improvements:
* escape the dot - otherwise two-digit major versions would be misinterpreted
* detect older version (for legacy behavior) instead of detecting new version: otherwise future versions (8.x) would be misinterpreted as old

quotengrote commented:
Good idea, changed it in https://github.com/munin-monitoring/contrib/pull/1262#issuecomment-986209397.

Sephtex commented:
I adapted my code, lets see if it'll produce the graphs now

Sephtex commented:
with the adapted change, I don't see any values yet, and the graph doesn't get recreated either:
![image](https://user-images.githubusercontent.com/25793199/144837762-fb1200ff-cacc-4e49-882e-677890e22177.png)
```
$ munin-run mikrotik_system
...
multigraph memory_Sephtex_router
total_memory.value 256
free_memory.value 222
used_memory.value 34
...
$ munin-run mikrotik_system config
...
multigraph memory_graph_Sephtex_router
graph_title memory Sephtex_router
graph_vlabel MiB
graph_category mikrotik
graph_args -l 0
total_memory.label total memory
used_memory.label used memory
free_memory.label free memory
graph_info Total, Used & free RAM.
used_memory.critical 201
used_memory.warning 156
...
```
Ok so again a missmatch between the multigraph name. I'll fix this

Sephtex commented:
fixed the graph name, lets see later, but I think it should be good now

quotengrote commented:
Works for me.

> Am 06.12.2021 um 12:30 schrieb Younes Ichiche ***@***.***>:
> 
> ﻿
> fixed the graph name, lets see later, but I think it should be good now
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub, or unsubscribe.

quotengrote commented:
So, after testing for two days with the crs309, rb750gr3 and rb5009 it works without problems with all of them.

Sephtex commented:
Same, just checked and all graphs are showing the values now as expected :-)

quotengrote commented:
@Sephtex Do you think it can get merged? If so, we should ping !sumpfralle

Sephtex commented:
@quotengrote yes we can defensively merge it! @sumpfralle go ahead I would say! ;-)

sumpfralle commented:
Now I merged the branch (with a bit of manual conflict resolving) - thanks to both of you for your work and discussion!
